packages/dd-trace/src/debugger/devtools_client/condition.js

```diff
@@ -0,0 +1,237 @@
+'use strict'
+
+module.exports = compile
+
+const identifierRegex = /^[@a-zA-Z_$][\w$]*$/
+
+// The following identifiers have purposefully not been included in this list:
+// - The reserved words `this` and `super` as they can have valid use cases as `ref` values
+// - The literals `undefined`, `Infinity` as they can be useful as `ref` values, especially to check if a
+//   variable is `undefined`.
+// - The following future reserved words in older standards, as they can now be used safely:
+//   `abstract`, `boolean`, `byte`, `char`, `double`, `final`, `float`, `goto`, `int`, `long`, `native`, `short`,
+//   `synchronized`, `throws`, `transient`, `volatile`.
+const reservedWords = new Set([
+  // Reserved words
+  'break', 'case', 'catch', 'class', 'const', 'continue', 'debugger', 'default', 'delete', 'do', 'else', 'export',
+  'extends', 'false', 'finally', 'for', 'function', 'if', 'import', 'in', 'instanceof', 'new', 'null', 'return',
+  'switch', 'throw', 'true', 'try', 'typeof', 'var', 'void', 'while', 'with',
+
+  // Reserved in strict mode
+  'let', 'static', 'yield',
+
+  // Reserved in module code or async function bodies:
+  'await',
+
+  // Future reserved words
+  'enum',
+
+  // Future reserved words in strict mode
+  'implements', 'interface', 'package', 'private', 'protected', 'public',
+
+  // Litterals
+  'NaN'
+])
+
+// TODO: Consider storing some of these functions on `process` so they can be reused across probes
+function compile (node) {
+  if (node === null || typeof node === 'number' || typeof node === 'boolean' || typeof node === 'string') {
+    return JSON.stringify(node)
+  }
+
+  const [type, value] = Object.entries(node)[0]
+
+  if (type === 'not') {
+    return `!(${compile(value)})`
+  } else if (type === 'len' || type === 'count') {
+    return getSize(compile(value))
+  } else if (type === 'isEmpty') {
+    return `${getSize(compile(value))} === 0`
+  } else if (type === 'isDefined') {
+    return `(() => {
+      try {
+        ${value}
+        return true
+      } catch {
+        return false
+      }
+    })()`
+  } else if (type === 'instanceof') {
+    return `Function.prototype[Symbol.hasInstance].call(${value[1]}, ${compile(value[0])})`
+  } else if (type === 'ref') {
+    if (value === '@it') {
+      return '$dd_it'
+    } else if (value === '@key') {
+      return '$dd_key'
+    } else if (value === '@value') {
+      return '$dd_value'
+    } else {
+      return assertIdentifier(value)
+    }
+  } else if (Array.isArray(value)) {
+    const args = value.map(compile)
+    switch (type) {
```

Comment:
Is this going to be a hot code path?

If yes, we could check for the length of each string to skip many checks in average (length === 2 || length === 3 ...).

Reply:
No it's only executed when a new probe is being added, which happens rarely. Maybe on a really active project it happens a few hundred times a day